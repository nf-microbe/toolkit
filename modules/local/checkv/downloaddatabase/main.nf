process CHECKV_DOWNLOADDATABASE {
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/checkv:1.0.1--pyhdfd78af_0':
        'biocontainers/checkv:1.0.1--pyhdfd78af_0' }"

    output:
    path "checkv_db/*"  , emit: checkv_db
    path "versions.yml" , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    """
    checkv download_database \\
        ${args} \\
        ./checkv_db

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        checkv: \$(checkv -h 2>&1  | sed -n 's/^.*CheckV v//; s/: assessing.*//; 1p')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''

    """
    mkdir checkv_db
    touch checkv_db/README.txt
    mkdir checkv_db/genome_db
    touch checkv_db/genome_db/changelog.tsv
    touch checkv_db/genome_db/checkv_error.tsv
    touch checkv_db/genome_db/checkv_info.tsv
    touch checkv_db/genome_db/checkv_reps.faa
    touch checkv_db/genome_db/checkv_reps.fna
    touch checkv_db/genome_db/checkv_reps.tsv
    mkdir checkv_db/hmm_db
    touch checkv_db/hmm_db/checkv_hmms.tsv
    touch checkv_db/hmm_db/genome_lengths.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        checkv: \$(checkv -h 2>&1  | sed -n 's/^.*CheckV v//; s/: assessing.*//; 1p')
    END_VERSIONS
    """

}
