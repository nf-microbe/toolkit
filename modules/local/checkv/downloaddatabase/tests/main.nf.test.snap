{
    "No input: download only - stub": {
        "content": [
            [
                "versions.yml:md5,29c4fbe7a6cb8b6c0913363d718cce9f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-10T13:41:55.53660563"
    },
    "No input: download only": {
        "content": [
            [
                "versions.yml:md5,29c4fbe7a6cb8b6c0913363d718cce9f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-10T13:41:35.511334063"
    }
}